lib/src/wallet/traits.rs

```diff
@@ -383,13 +398,100 @@ impl NoteAndMetadata for SaplingNoteAndMetadata {
         note.value
     }
 
+    fn spent(&self) -> &Option<(TxId, u32)> {
+        &self.spent
+    }
+
+    fn unconfirmed_spent(&self) -> &Option<(TxId, u32)> {
+        &self.unconfirmed_spent
+    }
+
+    fn witnesses(&mut self) -> &mut WitnessCache<Self::Node> {
+        &mut self.witnesses
+    }
+
     fn wallet_transaction_notes_mut(wallet_transaction: &mut WalletTx) -> &mut Vec<Self> {
         &mut wallet_transaction.sapling_notes
     }
 
+    fn wallet_transaction_notes(wallet_transaction: &WalletTx) -> &Vec<Self> {
+        &wallet_transaction.sapling_notes
+    }
+}
+
+impl NoteAndMetadata for OrchardNoteAndMetadata {
```

Comment:
Wait..  wasn't this already in place..  why does it seem to be added in this PR?

Reply:
This is moved from elsewhere in the same file, I'm pretty sure.